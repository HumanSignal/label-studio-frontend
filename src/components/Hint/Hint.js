import React, { Component } from "react";
import { PropTypes } from "prop-types";

import styles from "./Hint.module.scss";

/**
 * Hint Component
 * @param {object} style
 * @param {ant} children
 */
export default class Hint extends Component {
  render() {
    let style;
    let localClass = `${styles.main}`;

    if (this.props.style) style = this.props.style;

    if (this.props.className) {
      localClass = `${styles.main} ${this.props.className}`;
    }

    return (
      <sup data-copy={this.props.copy} className={localClass} style={style}>
        {this.props.children}
      </sup>
    );
  }
}

Hint.propTypes = {
  style: PropTypes.object,
  className: PropTypes.string,
  copy: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.array, PropTypes.string, PropTypes.object]),
};
