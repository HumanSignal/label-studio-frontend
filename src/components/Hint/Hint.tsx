import { Component, CSSProperties } from "react";
import { Block } from "../../utils/bem";

import "./Hint.styl";

interface HintProps {
  copy?: string;
  style?: CSSProperties;
  className?: string;
}

/**
 * Hint Component
 */
export default class Hint extends Component<HintProps> {
  render() {
    return (
      <Block
        name="hint"
        tag="sup"
        className={this.props.className}
        data-copy={this.props.copy}
        style={this.props.style}
      >
        {this.props.children}
      </Block>
    );
  }
}
