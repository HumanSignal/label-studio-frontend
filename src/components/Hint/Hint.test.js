/* global describe, it, expect */
import React from "react";
import Enzyme, { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import Hint from "./Hint";

describe("Hint", () => {
  it("Should render correctly", () => {
    const output = shallow(
      <Hint copy="test" style={{ background: "red" }} className="test">
        Test
      </Hint>,
    );

    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
