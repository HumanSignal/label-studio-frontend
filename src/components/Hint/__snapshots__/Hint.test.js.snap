// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hint Should render correctly 1`] = `
<Block
  className="test"
  data-copy="test"
  name="hint"
  style={
    Object {
      "background": "red",
    }
  }
  tag="sup"
>
  Test
</Block>
`;
